pkg/cluster/cluster.go

```diff
@@ -222,10 +218,15 @@ func (p *Peer) SetTimestamps(mint int64, maxt int64) {
 	p.data[p.Name()] = s
 }
 
-// Leave the cluster, waiting up to timeout.
-func (p *Peer) Leave(timeout time.Duration) error {
+// Close leaves the cluster waiting up to timeout and shutdowns peer if cluster left.
+// TODO(bplotka): Add this method into run.Group closing logic for each command. This will improve graceful shutdown.
+func (p *Peer) Close(timeout time.Duration) error {
+	err := p.mlist.Leave(timeout)
+	if err != nil {
+		return err
```

Comment:
I think that is the correct behavior isn't it? If leave fails, member is still in cluster.

Reply:
We can do your option as well, but then there is no point in returning error I think

Reply:
> I think that is the correct behavior isn't it? If leave fails, member is still in cluster.

I wondered that, but I'm not sure that's guaranteed either. The only reason I found for `Leave()` to return an error (in the current implementation) is due to a timeout. Which is why I think proceeding with the shutdown on error might be preferable.

> We can do your option as well, but then there is no point in returning error I think

Agreed. If you have access to a logger, it might be nice to log the error returned by `Leave()`, but I think we should continue to `Shutdown()`.